### PR TITLE
Fixes #7058: Fix reporting issues (mutiple reprtos for nexpanded variable) on branch 3.0

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/7058

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/domain/reports/StatusReports.scala [hunk 1]
```diff
@@ -276,11 +276,11 @@ object ComponentValueStatusReport extends Loggable {
...
    * them by component value
    */
-  def merge(values: Iterable[ComponentValueStatusReport]): Map[String, ComponentValueStatusReport] = {
-    val pairs = values.groupBy(_.componentValue).map { case (valueName, values) =>
+  def merge(values: Iterable[ComponentValueStatusReport]): Map[Option[String], ComponentValueStatusReport] = {
+    val pairs = values.groupBy(_.unexpandedComponentValue).map { case (unexpanded, values) =>
       //the unexpanded value should be the same on all values.
       //if not, report an error for devs
-      val (unexpanded, messages) = ((Option.empty[String],List.empty[MessageStatusReport])/:values) { case(acc,next) =>
+      /*val (unexpanded, messages) = ((Option.empty[String],List.empty[MessageStatusReport])/:values) { case(acc,next) =>
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/domain/reports/StatusReports.scala]: why not deleting this part directly ?

File: rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala [hunk 2]
```diff
@@ -799,95 +799,120 @@ object ExecutionBatch extends Loggable {
...
-      (Seq(), Seq())
-    } else {
-      val reports = purgedReports.filter(r => r.keyValue == "None")
-      (Seq(buildComponentValueStatus(
-          "None"
+    // Regroup all None value into None component Value
+    // There should only one None report per component and
+    val (noneValue, noneReports) = {
+      if(valueKind.none.isEmpty) {
+      (None, Seq())
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala]: wrong indentation

**Comment by ncharles** [comment]
this code would benefit from comments to explain what it does - future us would be so happy to read that

**Comment by fanf** [comment]
Superseeded by pr #909

**Pull request closed without merging**